### PR TITLE
feat: add custom dropdown menu item filter in edit mode

### PR DESCRIPTION
**Pull request by goshander**
implement new feature for filtering menu item in edit mode by custom `visible` handler

```tsx
type MenuItem = {
  visible?: (item: ConfigItem) => boolean; // optional visibility handler for filtering menu items
};
```

also fix dropdown menu item handler with `ConfigItem` as callback params

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,14 +57,15 @@ Before using `DashKit` as a react component, it must be configured.
...
-  Used for global DashKit settings (such as margins between widgets and default widget sizes)
+  Used for global DashKit settings (such as margins between widgets, default widget sizes and widget overlay menu)
 
   ```js
   import {DashKit} from '@yandex-cloud/dashkit';
 
   DashKit.setSettings({
     gridLayout: {margin: [8, 8]},
     isMobile: true,
+    // menu: [] as Array<MenuItem>,
```
Comment (jhoncool) [inline README.md:68]: Why is this commented out?

File: src/components/OverlayControls/OverlayControls.tsx [hunk 2]
```diff
@@ -209,13 +210,22 @@ class OverlayControls extends React.Component<OverlayControlsProps> {
...
                   }
+                  // custom menu dropdown item filter
+                  if (item.visible && !item.visible(this.props.configItem)) {
+                      return null;
+                  }
+
+                  const itemHandler = item.handler;
+
+                  const itemAction =
+                      itemHandler && typeof itemHandler === 'function'
```
Comment (jhoncool) [inline src/components/OverlayControls/OverlayControls.tsx:221]: Not need part with `itemHandler &&`.

**Review by jhoncool — COMMENTED** [review]
!

File: README.md (hunk 1 above)
Comment (goshander) [reply README.md:68]: This is just an example, if we really pass an empty array, we will get an empty menu

File: src/components/OverlayControls/OverlayControls.tsx (hunk 2 above)
Comment (goshander) [reply src/components/OverlayControls/OverlayControls.tsx]: fixed

**Review by jhoncool — APPROVED** [review]

**Pull request merged**